### DISCUSSION TITLE
How to open a Page in a new Window while maintaining UI state.

### DISCUSSION THREAD

**Discussion by technelak** [dotMorten/WinUIEx#116] [Q&A]
I've asked this question in another repo, but I wanted to get your take on it as well. 

What is the correct approach to open a Page in a new Window without losing its UI state? I have tried different methods but I am uncertain if I am on the right track. One attempt was to use a new Frame to navigate to the Page and set this.NavigationCacheMode = NavigationCacheMode.Required, but it did not work as expected. Also, a navigation state is not required for this page.

I am having difficulty copying the content of the page and setting a new Frame's content to the content of the original page without encountering a "Value does not fall within the expected range" error. Do I need to implement a service to save the state of each UI element on the page and reapply it when the page loads again? Or is there a simpler way to accomplish this?

**Comment by dotMorten** [comment]
What do you mean by “new window without losing UI state”?

**Reply by technelak** [reply]
For my application, I have the main Shell hosting a page within a Frame.  This page contains a Tabview with DataGrids and a few text boxes.  I want the user to be able to move this page into its own window while maintaining the state of the Tabview and textboxes i.e. any open tabs, column reordering/sorting, and any inputs in the textbox.   I'm able to capture and preserve the state of the ViewModel however I lose any open tabs or column sorting. Does that make more sense? I'm sorry if my explanation is a bit skewed. 

I'm trying to accomplish this example. https://learn.microsoft.com/en-us/windows/apps/design/layout/show-multiple-views

**Reply by dotMorten** [reply]
This sounds like a general WinUI question out of scope of WinUIEx
